### PR TITLE
improve coco_error_analysis.py

### PR DESCRIPTION
**Pull request by fcakyon**
bbox-allclass-ap bar plot
gt annotation areas histogram plot
number of annotations per area group

### CODE REVIEW

File: tools/analysis_tools/coco_error_analysis.py [hunk 1]
```diff
@@ -147,10 +201,13 @@ def analyze_results(res_file, ann_file, res_types, out_dir):
...
             # fill in background and false negative errors and plot
             ps[ps == -1] = 0
-            ps[5, :, k, :, :] = (ps[4, :, k, :, :] > 0)
+            ps[5, :, k, :, :] = ps[4, :, k, :, :] > 0
             ps[6, :, k, :, :] = 1.0
             makeplot(recThrs, ps[:, :, k], res_out_dir, nm['name'], iou_type)
+            makebarplot(recThrs, ps[:, :, k], res_out_dir, nm['name'],
+                        iou_type)
         makeplot(recThrs, ps, res_out_dir, 'allclass', iou_type)
+        makebarplot(recThrs, ps, res_out_dir, 'allclass', iou_type)
```
Comment (ZwwWayne) [inline tools/analysis_tools/coco_error_analysis.py:210]: Suggest to make it as an option

Comment (fcakyon) [reply tools/analysis_tools/coco_error_analysis.py]: made all the changes optional

File: tools/analysis_tools/coco_error_analysis.py [hunk 2]
```diff
@@ -37,19 +37,149 @@ def makeplot(rs, ps, outDir, class_name, iou_type):
...
+    # Init figure
+    fig, ax = plt.subplots()
+    x = np.arange(len(areaRngLbl))  # the areaNames locations
+    width = 0.60  # the width of the bars
+    figure_title = 'number of annotations per area group'
+
+    rects = ax.bar(
+        x,
+        areaRngLbl2Number.values(),
+        width,
```
Comment (ZwwWayne) [inline tools/analysis_tools/coco_error_analysis.py:137]: clean unnecessary comma

File: tools/analysis_tools/coco_error_analysis.py [hunk 3]
```diff
@@ -160,11 +315,28 @@ def main():
     parser.add_argument(
         '--ann',
         default='data/coco/annotations/instances_val2017.json',
-        help='annotation file path')
+        help='annotation file path',
```
Comment (ZwwWayne) [inline tools/analysis_tools/coco_error_analysis.py:318]: clean comma.

**Comment by ZwwWayne** [comment]
LGTM after resolving the final comments.

File: tools/analysis_tools/coco_error_analysis.py (hunk 3 above)
Comment (fcakyon) [reply tools/analysis_tools/coco_error_analysis.py]: @ZwwWayne done

File: tools/analysis_tools/coco_error_analysis.py (hunk 2 above)
Comment (fcakyon) [reply tools/analysis_tools/coco_error_analysis.py]: @ZwwWayne done

**Review by hhaAndroid — APPROVED** [review]

**Comment by fcakyon** [comment]
any news? @ZwwWayne @jshilong

**Comment by OpenMMLab-Miao** [comment]
Dear fcakyon,
First of all, we want to express our gratitude for your significant PR in the MMDet project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.
If you are Chinese or have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. We look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG
Thank you again for your contribution❤
Best regards！ @fcakyon

**Pull request merged**